samples/stock/demo/orderbook-chart/demo.js

```diff
@@ -0,0 +1,187 @@
+Highcharts.Templating.helpers.abs = value => Math.abs(value);
+
+function getRandomNumber(min, max) {
+    return Math.round(Math.random() * (max - min)) + min;
+}
+
+function generateBidAndAskData(n) {
+    const data = [[], []];
+    let bidPrice = getRandomNumber(29000, 30000),
+        askPrice = bidPrice + 0.5;
+    for (let i = 0; i < n; i++) {
+        bidPrice -= ((i * getRandomNumber(8, 10)));
+        askPrice += ((i * getRandomNumber(8, 10)));
+        data[0].push({
+            x: i,
+            y: (-i - 1) * getRandomNumber(70000, 110000),
+            price: bidPrice
+        });
+
+        data[1].push({
+            x: i,
+            y: (i + 1) * getRandomNumber(70000, 110000),
+            price: askPrice
+        });
+    }
+    return data;
+}
+
+const [bidsData, asksData] = generateBidAndAskData(10);
+
+function updateData(chart) {
+    const data = generateBidAndAskData(10);
+    chart.series.forEach((s, i) => {
+        s.setData(data[i], false);
+    });
+    chart.redraw();
+}
+
+Highcharts.chart('container', {
+    chart: {
+        animation: {
+            duration: 200
+        },
+        type: 'bar',
+        backgroundColor: '#23232f',
+        marginTop: 70,
+        events: {
+            load() {
+                setInterval(() => {
+                    if (this.series) {
+                        updateData(this);
+                    }
+                }, 200);
+            }
+        }
+    },
+
+    title: {
+        text: 'Order book live chart',
+        style: {
+            color: '#ffffff'
+        }
+    },
+
+    tooltip: {
+        headerFormat: 'Price: <b>${point.point.price:,.1f}</b></br>',
+        pointFormat: '{series.name}: <b>{(abs point.y):,.0f}</b>',
+        shape: 'rect',
+        positioner(labelWidth, _, point) {
+            const { plotX, plotY, h, negative } = point;
+            return {
+                x: negative ? plotX + h - labelWidth : plotX - h,
+                y: plotY
+            };
+        }
+    },
+
+    xAxis: [{
+        reversed: true,
+        visible: false,
+        title: {
+            text: 'Market depth / price'
+        }
+    }, {
+        opposite: true,
+        visible: false,
+        title: {
+            text: 'Market depth / price'
+        }
+    }],
+
+    yAxis: {
```

Comment:
Instead of the hack with assigning negative values and displaying them as positive using the `abs` helper, it would be more appropriate to use two y-axes. One reversed, the other one plain, and positioned using `left` and `width` option.

I appreciate that the current approach is borrowed from the `bar-negative-stack` demo, but that was created before we supported a percentage left/width, and does not currently reflect best practice. We are planning to fix that.

Reply:
@TorsteinHonsi Thanks! I have implemented your suggestions and layout came out nicely, however I'm afraid I have discovered a new bug 😬 

https://jsfiddle.net/BlackLabel/bsmxz2n3/

The dataLabels on the right yAxis fall out of the chart, if the `yAxis.left` is greater than 0. It looks like it could have something to do with the `justifyDataLabel`? https://github.com/highcharts/highcharts/blob/81ce84997857d27756365ec2b2c05bd3c89c64a1/ts/Core/Series/DataLabel.ts#L867-L877

I haven't debugged it, but quick look into the method leads to conclusion that maybe it should take an axis width/len, instead of the `chart.plotWidth`.

What do you think about it? Is it actually a bug, and should we fix it - or the core team should handle it?

Reply:
You're probably right about it using `plotWidth` instead of `axis.pos + axis.len`. 

> What do you think about it? Is it actually a bug, and should we fix it - or the core team should handle it?

I think you should set aside a reasonable amount of time trying to fix it - say two hours? If it turns out to be a Pandora's box, then report back, so we can decide whether to invest more in it. Otherwise we're all happy.